snapcraft/plugins/go.py

```diff
@@ -101,7 +102,13 @@ def get_pull_properties(cls):
 
     def __init__(self, name, options, project):
         super().__init__(name, options, project)
-        self.build_packages.append('golang-go')
+
+        # Do not try to install golang if it is already available in our PATH.
+        try:
+            subprocess.check_output(['go', 'version'])
+        except OSError:
+            self.build_packages.append('golang-go')
```

Comment:
Thanks for your contribution @chrisglass. However, our plans seem to go in the opposite direction.

We want snapcraft to work on a clean xenial container, where go will not be installed. We are trying to default to build in containers always, and to make reproducible builds. Relying in the go package installed in the developer machine goes against this.

What we want is to support build-snaps, so you can specify on the yaml the snap with your toolchain. If your machine already has this snap installed, then nothing will happen. And this also encourages people to push to the store snaps for compilers and stuff like this.

I don't think we want to support a case for a developer installing his own toolchain from something that's not a snap, but this is open to debate of course. You can open a forum post to discuss with a wider audience.

Please let me know what do you think.